### PR TITLE
dm: support dump task mode in openapi

### PR DESCRIPTION
**Pull request by River2000i**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/9230

### What is changed and how it works?
1. generate go client and server by `make dm_generate_openapi `
2. add opanapi test from dump task and load&sync task

#### Tests 

 - [ ] Unit test
 - [x] Integration test
 - [ ] Manual test (add detailed scripts or steps below)
 - [ ] No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

**Comment by River2000i** [comment]
@GMHDBJD @lance6716 @D3Hunter PTAL~

File: dm/openapi/spec/dm.yaml [hunk 1]
```diff
@@ -1825,6 +1825,8 @@ components:
             - "full"
             - "incremental"
             - "all"
+            - "dump"
+            - "load&sync"
```
Comment (D3Hunter) [inline dm/openapi/spec/dm.yaml:1829]: no need to expose this one

Comment (River2000i) [reply dm/openapi/spec/dm.yaml:1829]: get it, i will revert relate code and test.

Comment (River2000i) [reply dm/openapi/spec/dm.yaml]: done

**Comment by River2000i** [comment]
ut https://github.com/pingcap/tiflow/issues/11361  failed, rerun it

File: dm/tests/openapi/run.sh [hunk 2]
```diff
@@ -176,6 +182,52 @@ function test_relay() {
...
+
+	init_dump_data
+
+	# start dump task success
+	openapi_task_check "start_task_success" $task_name ""
+	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"query-status $task_name" \
+		"\"stage\": \"Running\"" 1
+
+	# wait dump task finish
```
Comment (GMHDBJD) [inline dm/tests/openapi/run.sh:221]: can you check it with get_task_status_check via openapi instead of dmctl?

**Review by GMHDBJD — APPROVED** [review]
rest LGTM

**Review by D3Hunter — APPROVED** [review]

File: dm/tests/openapi/run.sh (hunk 2 above)
Comment (River2000i) [reply dm/tests/openapi/run.sh:221]: Validating json fields in python is a bit complex, I'll create an issue to improve it later...

**Pull request merged**